src/daft-dsl/src/python.rs

```diff
@@ -71,6 +73,55 @@ pub fn lit(item: &PyAny) -> PyResult<PyExpr> {
     } else if let Ok(pybytes) = item.downcast::<PyBytes>() {
         let bytes = pybytes.as_bytes();
         Ok(crate::lit(bytes).into())
+    } else if let Ok(pylist) = item.downcast::<PyList>() {
```

Comment:
See comment above about using a SeriesLiteral. You can then avoid all this parsing code. Espically if you end up with some input like `[False, 1, 1.0]`